packages/next/client/image.tsx

```diff
@@ -234,6 +237,11 @@ export default function Image({
     lazy = true
   }
 
+  if (typeof window !== 'undefined' && !window.IntersectionObserver) {
```

Comment:
You're doing this check twice, can you do it once and pass in the variable to the observer?

Reply:
That's actually a little trickier than you would think because of the weird test for missing intersection observer. Re-accessing the window.IntersectionObserver at the time of getObserver allows the test to delete that property in time. If I cache it at the root level of the script, it doesn't work.